src/commands/deployWorkspaceProject/getDefaultValues/getDefaultContainerAppsResources.ts

```diff
@@ -56,3 +67,38 @@ export async function getDefaultContainerAppsResources(context: ISubscriptionAct
         return noMatchingResources;
     }
 }
+
+async function promptForContainerAppsEnvironmentResources(context: ISubscriptionActionContext): Promise<DefaultContainerAppsResources> {
+    const client: ContainerAppsAPIClient = await createContainerAppsAPIClient(context)
+    const managedEnvironments: ManagedEnvironment[] = await uiUtils.listAllIterator(client.managedEnvironments.listBySubscription());
+
+    if (!managedEnvironments.length) {
+        return noMatchingResources;
+    }
+
+    const picks: IAzureQuickPickItem<ManagedEnvironment | undefined>[] = [
+        {
+            label: localize('newManagedEnvironment', '$(plus) Create new container apps environment'),
+            description: '',
+            data: undefined
+        },
+        ...managedEnvironments.map(env => {
+            return {
+                label: nonNullProp(env, 'name'),
+                description: '',
+                data: env
+            };
+        })
+    ];
+
+    const placeHolder: string = localize('selectManagedEnvironment', 'Select a container apps environment');
+    const managedEnvironment: ManagedEnvironment | undefined = (await context.ui.showQuickPick(picks, { placeHolder })).data;
+
+    if (!managedEnvironment) {
```

Comment:
Wouldn't `managedEnvironment` be undefined here if the user selected `create new container app environment`? I'm thoroughly confused on what is going on here.

Like you said in your PR, I feel like this logic should be in a `ManagedEnvironmentListStep` rather than having it be a part of the `getDefaultContainerAppsResources`.

My opinion is that this command should _only_ try to get the default values. If it can't, it exits out and we rely on the wizard prompts to get input from the user.

Reply:
Based on the comment above, and this comment, I think the confusion is due to thinking about this as if it were a list step being used as part of our Azure Wizard flow.  

> Wouldn't managedEnvironment be undefined here if the user selected create new container app environment?
In this scenario, `undefined`
Yes, exactly this, if the user selects create, by setting as `undefined`, we are indicating that no default resource exists and that one needs to be created later.

> Like you said in your PR, I feel like this logic should be in a ManagedEnvironmentListStep rather than having it be a part of the getDefaultContainerAppsResources.

I thought about making it all as a list step, but there's so much other stuff I would then have to build out and think about when all I really need is the `getPicks` portion of the list step essentially... so I decided to defer creating a list step until I actually need to use one.